detekt-rules/src/main/kotlin/OrtPackageNaming.kt

```diff
@@ -58,7 +58,11 @@ class OrtPackageNaming(config: Config) : Rule(config) {
 
         val (pathPrefix, pathSuffix) = path.split(pathPattern, 2).map { File(it) }
         val projectDir = pathPrefix.name
-        val projectGroup = pathPrefix.parent?.let { ".$it" }.orEmpty()
+        val projectGroup = pathPrefix.parent
+            ?.let { ".$it" }
+            ?.replace("/", ".")
+            ?.replace("-", "")
```

Comment:
Nit: Could use the `Char`-version of `replace()` by using single quotes.

Reply:
See comment in #6573.

Reply:
I don't find a comment there, but by now I guess you kept it at strings as you need to replace with an empty string which is not possible with chars.